### PR TITLE
chore: added logs in update action flow

### LINKED ISSUE #30282: [Task]: Add more info logs for update action API
### SubTasks

While looking into a-force [issue](https://github.com/appsmithorg/appsmith/issues/30102), the page where the action was failing was corrupted somehow and we could not even figure that out through logs, only thing that was clear was that update action API was returning 200 OK response with no payload, consequently it would fail to call the execute API and would remain in the loading state. The logs that user provided were not very useful as logs were not enough. Update action API does not have enough logs in case of any failures which makes it really difficult to pin point the problem when issue is not replicable locally, hence we should add more logs.

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR adds logs to update action flow, so that we can debug a-force issues faster.

#### PR fixes following issue(s)
Fixes #30282
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Chores**
	- Enhanced logging for better debugging and monitoring of action updates.
	- Optimized import statements in test files for improved code maintainability.
	- Refactored code for improved performance.

### CODE REVIEW

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ApplicationServiceCETest.java [hunk 1]
```diff
@@ -30,7 +30,6 @@
 import com.appsmith.server.domains.NewPage;
 import com.appsmith.server.domains.PermissionGroup;
 import com.appsmith.server.domains.Plugin;
-import com.appsmith.server.domains.QNewAction;
```
Comment (sneha122) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/ApplicationServiceCETest.java:33]: Changes in this file are done automatically when I ran `mvn spotless:apply` on server code base

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/ResponseUtilsCE.java [hunk 2]
```diff
@@ -105,6 +105,7 @@ public ApplicationPagesDTO updateApplicationPagesDTOWithDefaultResources(Applica
     }
 
     public ActionDTO updateActionDTOWithDefaultResources(ActionDTO action) {
+        log.debug("Updating action DTO with default resources");
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/ResponseUtilsCE.java:108]: Can we please improve this log message by printing the ID of the action object? Without the action ID, it's really hard to understand the flow of a particular action that we wish to debug. 

Please do this across the board. As an exercise, once you make the code changes, run the API 10 times in quick succession. Then look at **_only_** the logs & see if you can understand what's happening by using the logs. This simple 30secs approach goes a long way in determining if logs being added are useful or not.

Comment (sneha122) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/ce/ResponseUtilsCE.java:108]: Makes absolute sense!! I will make this change across all the logs

**Review by mohanarpit — APPROVED** [review]

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**